### DISCUSSION TITLE
KV: partial mutation for object

### DISCUSSION THREAD

**Discussion by sylc** [denoland/deno#19026] [Ideas]
Hi, 

After a bit of play with Deno KV, I found myself wishing sometimes there was an api to mutate an object property, without having to provide the full object, something like a partial mutation. 

currently

```ts
// at creation time
const config = { setting1: 'a', setting 2: 'b' }
await kv.set(["config"], config )

// later on in the application, need to change setting 1 only
const config = await kv.get(["config"])
await kv.set(["config"], { ...config.value, setting1: 'z'})
```

New API suggestion
```ts
await kv.atomic().mutate({ key: ["config"], type: 'partial' ; value: { setting1: 'z' }).commit()
// alias kv.partial(["config"], { setting1: 'z' })
```

Although I'm not sure if this is a good idea as it could maybe be anti pattern because one work around is to split the config with multiple keys, like below. But this approach feels less flexible and verbose at first.

```ts
// at creation
await kv.atomic
  .set(["config", 'setting1'], 'a')
  .set(["config", 'setting2'], 'b')
  .atomic()

// later on
await kv.set(["config", 'setting1'], 'a')
```

**Comment by ajinkyashukla** [comment]
This is much needed. It is inefficient to pass the whole object over the network to update just one attribute.

Any update on this Deno team ?
